src/main/java/dtb/lego/truck/rest/data/events/acquisition/boundary/EventHistoryResource.java

```diff
@@ -0,0 +1,81 @@
+            "Set 0 on end to use the current time instead")
+    @GetMapping("/events/history")
+    public ResponseEntity<Collection<? extends Event>> getComponentEventHistoryInInterval(@RequestParam String component,
+                                                                                          @RequestParam(name = "from") @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) ZonedDateTime from,
+                                                                                          @RequestParam(name = "to") @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) ZonedDateTime to) {
+        long begin = from.toInstant().toEpochMilli();
+        long end = to.toInstant().toEpochMilli();
+        inputValidator.checkValidComponent(component);
+        inputValidator.checkThatBeginIsBeforeEnd(begin, end);
+
+        Collection<? extends Event> events = sensorDataHandler.getEventsInInterval(component, begin, end);
+        return ResponseEntity.ok()
+                .body(events);
+    }
+
+    //2019-08-06T13:55:31.016+01:00
+    @ApiOperation(value = "Get all the events from the component contained in the requested interval, with the sampling rate requested" +
+            " and applying the required transformation")
+    @GetMapping("/events/history/transformed")
+    public ResponseEntity<Collection<? extends Event>> getTransformedComponentEventHistoryInInterval(@RequestParam String component,
+                                                                                                     @RequestParam(name = "from") @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) ZonedDateTime from,
+                                                                                                     @RequestParam(name = "to") @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) ZonedDateTime to,
+                                                                                                     @RequestParam int sampling,
+                                                                                                     @RequestParam String transformation,
+                                                                                                     @RequestParam(defaultValue = "false") boolean fillGaps) {
+        long begin = from.toInstant().toEpochMilli();
+        long end = to.toInstant().toEpochMilli();
+        inputValidator.checkValidComponent(component);
+        inputValidator.checkThatBeginIsBeforeEnd(begin, end);
+        inputValidator.checkBeginTransformedHistoryLessThan1MonthAgo(begin);
+        inputValidator.checkValidTransformation(transformation);
+
+        Collection<? extends Event> events = sensorDataHandler.getTransformedEventHistory(sampling, component, transformation, begin, end, fillGaps);
+        return ResponseEntity.ok()
+                .body(events);
+    }
+
+    @DeleteMapping("/events/history")
```

Comment:
Maybe makes sense to create an /admin resource and move the delete to there

Reply:
Should I create a new RestController for the admin resource into the same package?